ui/src/app/applications/components/application-retry-options/application-retry-options.tsx

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+import {FormApi, Text} from 'react-form';
+import {FormField} from 'argo-ui';
+import {NumberField} from '../../../shared/components';
+
+require('./application-retry-options.scss');
+
+
+
+function buildFormItem(label: string, propertyPath: string, component: React.ComponentType, formApi: FormApi, componentProps?: Record<string, any>) {
+    return <FormField
+        formApi={formApi}
+        label={label}
+        field={propertyPath}
+        component={component}
+        componentProps={componentProps}
+    />
+}
+
+const onlyPositiveValidation = {
+    min:'1',
+    step:'1'
+}
+
+const retryOptions: Array<(formApi: FormApi) => React.ReactNode> = [
+    formApi => buildFormItem('Limit', 'retryStrategy.limit', NumberField, formApi, onlyPositiveValidation),
+    formApi => buildFormItem('Duration', 'retryStrategy.backoff.duration', Text, formApi),
+    formApi => buildFormItem('Max Duration', 'retryStrategy.backoff.maxDuration', Text, formApi),
+    formApi => buildFormItem('Factor', 'retryStrategy.backoff.factor', NumberField, formApi, onlyPositiveValidation),
+];
+
+
+export const ApplicationRetryOptions = ({ formApi }: { formApi: FormApi }) => {
+
+    return <div className='row application-retry-options'>
```

Comment:
The user should be able to enabled or disable retry. Please add a checkbox that enables/disables retry. The retry parameters panel should be shown only if checkbox is checked.